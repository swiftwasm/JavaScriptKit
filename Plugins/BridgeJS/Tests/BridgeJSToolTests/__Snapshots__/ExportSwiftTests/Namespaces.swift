// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `swift package bridge-js`.

@_spi(BridgeJS) import JavaScriptKit

@_expose(wasm, "bjs_plainFunction")
@_cdecl("bjs_plainFunction")
public func _bjs_plainFunction() -> Void {
    #if arch(wasm32)
    let ret = plainFunction()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_MyModule_Utils_namespacedFunction")
@_cdecl("bjs_MyModule_Utils_namespacedFunction")
public func _bjs_MyModule_Utils_namespacedFunction() -> Void {
    #if arch(wasm32)
    let ret = namespacedFunction()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Greeter_init")
@_cdecl("bjs_Greeter_init")
public func _bjs_Greeter_init(nameBytes: Int32, nameLength: Int32) -> UnsafeMutableRawPointer {
    #if arch(wasm32)
    let ret = Greeter(name: String.bridgeJSLiftParameter(nameBytes, nameLength))
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Greeter_greet")
@_cdecl("bjs_Greeter_greet")
public func _bjs_Greeter_greet(_self: UnsafeMutableRawPointer) -> Void {
    #if arch(wasm32)
    let ret = Greeter.bridgeJSLiftParameter(_self).greet()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Greeter_deinit")
@_cdecl("bjs_Greeter_deinit")
public func _bjs_Greeter_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<Greeter>.fromOpaque(pointer).release()
}

extension Greeter: ConvertibleToJSValue, _BridgedSwiftHeapObject {
    var jsValue: JSValue {
        #if arch(wasm32)
        @_extern(wasm, module: "TestModule", name: "bjs_Greeter_wrap")
        func _bjs_Greeter_wrap(_: UnsafeMutableRawPointer) -> Int32
        #else
        func _bjs_Greeter_wrap(_: UnsafeMutableRawPointer) -> Int32 {
            fatalError("Only available on WebAssembly")
        }
        #endif
        return .object(JSObject(id: UInt32(bitPattern: _bjs_Greeter_wrap(Unmanaged.passRetained(self).toOpaque()))))
    }
}

@_expose(wasm, "bjs_Converter_init")
@_cdecl("bjs_Converter_init")
public func _bjs_Converter_init() -> UnsafeMutableRawPointer {
    #if arch(wasm32)
    let ret = Converter()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Converter_toString")
@_cdecl("bjs_Converter_toString")
public func _bjs_Converter_toString(_self: UnsafeMutableRawPointer, value: Int32) -> Void {
    #if arch(wasm32)
    let ret = Converter.bridgeJSLiftParameter(_self).toString(value: Int.bridgeJSLiftParameter(value))
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Converter_deinit")
@_cdecl("bjs_Converter_deinit")
public func _bjs_Converter_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<Converter>.fromOpaque(pointer).release()
}

extension Converter: ConvertibleToJSValue, _BridgedSwiftHeapObject {
    var jsValue: JSValue {
        #if arch(wasm32)
        @_extern(wasm, module: "TestModule", name: "bjs_Converter_wrap")
        func _bjs_Converter_wrap(_: UnsafeMutableRawPointer) -> Int32
        #else
        func _bjs_Converter_wrap(_: UnsafeMutableRawPointer) -> Int32 {
            fatalError("Only available on WebAssembly")
        }
        #endif
        return .object(JSObject(id: UInt32(bitPattern: _bjs_Converter_wrap(Unmanaged.passRetained(self).toOpaque()))))
    }
}

@_expose(wasm, "bjs_UUID_uuidString")
@_cdecl("bjs_UUID_uuidString")
public func _bjs_UUID_uuidString(_self: UnsafeMutableRawPointer) -> Void {
    #if arch(wasm32)
    let ret = UUID.bridgeJSLiftParameter(_self).uuidString()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_UUID_deinit")
@_cdecl("bjs_UUID_deinit")
public func _bjs_UUID_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<UUID>.fromOpaque(pointer).release()
}

extension UUID: ConvertibleToJSValue, _BridgedSwiftHeapObject {
    var jsValue: JSValue {
        #if arch(wasm32)
        @_extern(wasm, module: "TestModule", name: "bjs_UUID_wrap")
        func _bjs_UUID_wrap(_: UnsafeMutableRawPointer) -> Int32
        #else
        func _bjs_UUID_wrap(_: UnsafeMutableRawPointer) -> Int32 {
            fatalError("Only available on WebAssembly")
        }
        #endif
        return .object(JSObject(id: UInt32(bitPattern: _bjs_UUID_wrap(Unmanaged.passRetained(self).toOpaque()))))
    }
}