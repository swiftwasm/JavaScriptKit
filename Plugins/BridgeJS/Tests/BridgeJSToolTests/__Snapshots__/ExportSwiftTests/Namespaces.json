{
  "classes" : [
    {
      "constructor" : {
        "abiName" : "bjs_Greeter_init",
        "effects" : {
          "isAsync" : false,
          "isStatic" : false,
          "isThrows" : false
        },
        "parameters" : [
          {
            "label" : "name",
            "name" : "name",
            "type" : {
              "string" : {

              }
            }
          }
        ]
      },
      "methods" : [
        {
          "abiName" : "bjs_Greeter_greet",
          "effects" : {
            "isAsync" : false,
            "isStatic" : false,
            "isThrows" : false
          },
          "name" : "greet",
          "parameters" : [

          ],
          "returnType" : {
            "string" : {

            }
          }
        }
      ],
      "name" : "Greeter",
      "namespace" : [
        "__Swift",
        "Foundation"
      ],
      "properties" : [

      ],
      "swiftCallName" : "Greeter"
    },
    {
      "constructor" : {
        "abiName" : "bjs_Converter_init",
        "effects" : {
          "isAsync" : false,
          "isStatic" : false,
          "isThrows" : false
        },
        "parameters" : [

        ]
      },
      "methods" : [
        {
          "abiName" : "bjs_Converter_toString",
          "effects" : {
            "isAsync" : false,
            "isStatic" : false,
            "isThrows" : false
          },
          "name" : "toString",
          "parameters" : [
            {
              "label" : "value",
              "name" : "value",
              "type" : {
                "int" : {

                }
              }
            }
          ],
          "returnType" : {
            "string" : {

            }
          }
        }
      ],
      "name" : "Converter",
      "namespace" : [
        "Utils",
        "Converters"
      ],
      "properties" : [

      ],
      "swiftCallName" : "Converter"
    },
    {
      "methods" : [
        {
          "abiName" : "bjs_UUID_uuidString",
          "effects" : {
            "isAsync" : false,
            "isStatic" : false,
            "isThrows" : false
          },
          "name" : "uuidString",
          "parameters" : [

          ],
          "returnType" : {
            "string" : {

            }
          }
        }
      ],
      "name" : "UUID",
      "namespace" : [
        "__Swift",
        "Foundation"
      ],
      "properties" : [

      ],
      "swiftCallName" : "UUID"
    }
  ],
  "enums" : [

  ],
  "functions" : [
    {
      "abiName" : "bjs_plainFunction",
      "effects" : {
        "isAsync" : false,
        "isStatic" : false,
        "isThrows" : false
      },
      "name" : "plainFunction",
      "parameters" : [

      ],
      "returnType" : {
        "string" : {

        }
      }
    },
    {
      "abiName" : "bjs_MyModule_Utils_namespacedFunction",
      "effects" : {
        "isAsync" : false,
        "isStatic" : false,
        "isThrows" : false
      },
      "name" : "namespacedFunction",
      "namespace" : [
        "MyModule",
        "Utils"
      ],
      "parameters" : [

      ],
      "returnType" : {
        "string" : {

        }
      }
    }
  ],
  "moduleName" : "TestModule",
  "protocols" : [

  ]
}